dagshub/data_engine/model/query_result.py

```diff
@@ -59,6 +59,10 @@
 logger = logging.getLogger(__name__)
 
 
+class CustomPredictor(Protocol):
+    def __call__(self, local_paths: List[str]) -> List[Tuple[Any, Optional[float]]]: ...
+
```

Comment:
In my experience Protocols are still not really picked up in some IDEs, plus additionally this is only one function, so you can make this a type variable instead:
```
CustomPredictor = Callable[List[str], List[Tuple[Any, Optional[float]]]
```